### PR TITLE
feat: improve environment creation and refine iers manager

### PR DESCRIPTION
**Pull request by apaletta3**

### CODE REVIEW

File: docker/development/debian/Dockerfile [hunk 1]
```diff
@@ -85,11 +85,11 @@ RUN cd /tmp \
  && apt-get remove -y csh \
  && rm -rf /var/lib/apt/lists/*
 
-# GeographicLib [1.49]
+# GeographicLib [r1.52]
```
Comment (Derollez) [inline docker/development/debian/Dockerfile:88]: @apaletta3 Update the READMe accordingly with new version?

File: test/OpenSpaceToolkit/Physics/Environment.test.cpp [hunk 2]
```diff
@@ -753,3 +757,144 @@ TEST (OpenSpaceToolkit_Physics_Environment, Test_3)
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+TEST (OpenSpaceToolkit_Physics_Environment, EarthCreationTest)
```
Comment (Derollez) [inline test/OpenSpaceToolkit/Physics/Environment.test.cpp:761]: rename test: We are testing the creation of an `Environment` using Earth Model only but this function should focus on ensuring the creator for `Environment` is functional!

File: test/OpenSpaceToolkit/Physics/Environment.test.cpp [hunk 3]
```diff
@@ -753,3 +757,144 @@ TEST (OpenSpaceToolkit_Physics_Environment, Test_3)
...
+        const Vector gravField = earthCelestial->getGravitationalFieldAt(Position::Meters(Vector3d(6371000.0,0.0,0.0), Frame::ITRF())) ;
+        const Vector3d expectedGravField = {-398600441500000.0 / 40589641000000,0.0,0.0 } ;
+
+        EXPECT_EQ( gravField.getValue(), expectedGravField ) ;
+        EXPECT_EQ(earthCelestial->getGravitationalParameter(), Derived(398600441500000.0, GravitationalParameterSIUnit)) ;
+        EXPECT_EQ(earthCelestial->getJ2(), 0.0) ;
+        EXPECT_EQ(earthCelestial->getJ4(), 0.0) ;
+        EXPECT_EQ(earthCelestial->getFlattening(), 1.0) ;
+
+
```
Comment (Derollez) [inline test/OpenSpaceToolkit/Physics/Environment.test.cpp:845]: remove extra blank lines

File: test/OpenSpaceToolkit/Physics/Environment.test.cpp [hunk 4]
```diff
@@ -753,3 +757,144 @@ TEST (OpenSpaceToolkit_Physics_Environment, Test_3)
...
+    using ostk::physics::env::obj::Celestial ;
+    using ostk::physics::env::obj::celest::Earth ;
+    using ostk::physics::coord::Frame ;
+    using ostk::physics::coord::Position ;
+    using ostk::physics::data::Vector ;
+
+
+    {
+        static const Derived::Unit      GravitationalParameterSIUnit                    =   Derived::Unit::GravitationalParameter(Length::Unit::Meter, Time::Unit::Second) ;
+        // const Derived                   Earth::Models::Spherical::GravitationalParameter=       { 398600441500000.0, GravitationalParameterSIUnit } ;
```
Comment (Derollez) [inline test/OpenSpaceToolkit/Physics/Environment.test.cpp:823]: TBR?

File: test/OpenSpaceToolkit/Physics/Environment.test.cpp [hunk 5]
```diff
@@ -753,3 +757,144 @@ TEST (OpenSpaceToolkit_Physics_Environment, Test_3)
...
+
+    }
+
+    // [TBI] Add a test for the relevant parameters of each earth model at different levels of truncations
+
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+TEST (OpenSpaceToolkit_Physics_Environment, SphericalEarthTest)
```
Comment (Derollez) [inline test/OpenSpaceToolkit/Physics/Environment.test.cpp:795]: Same test as the previous one but with a different Earth model argument correct? If so, should be a additional scope inside the same test function IMO

File: test/OpenSpaceToolkit/Physics/Environment.test.cpp [hunk 6]
```diff
@@ -753,3 +757,144 @@ TEST (OpenSpaceToolkit_Physics_Environment, Test_3)
...
+        const Array<Shared<Object>> objects =
+        {
+            std::make_shared<Earth>(Earth::Spherical())
+        } ;
+
+        Environment environment(instant, objects) ;
+
+        Shared<const Celestial> earthCelestial = environment.accessCelestialObjectWithName("Earth") ;
+
+        const Vector gravField = earthCelestial->getGravitationalFieldAt(Position::Meters(Vector3d(6371000.0,0.0,0.0), Frame::ITRF())) ;
```
Comment (Derollez) [inline test/OpenSpaceToolkit/Physics/Environment.test.cpp:836]: These lines are testing functions on `Celestial` and `Earth` specifically and are not specific to `Environment`, correct?

File: test/OpenSpaceToolkit/Physics/Environment.test.cpp [hunk 7]
```diff
@@ -753,3 +757,144 @@ TEST (OpenSpaceToolkit_Physics_Environment, Test_3)
...
+    using ostk::physics::coord::Position ;
+    using ostk::physics::data::Vector ;
+
+    {
+
+        const Instant instant = Instant::J2000() ;
+
+        const Array<Shared<Object>> objects =
+        {
+            std::make_shared<Earth>(Earth::EGM84(2,2))
```
Comment (Derollez) [inline test/OpenSpaceToolkit/Physics/Environment.test.cpp:884]: Same as above

Comment (Derollez) [reply test/OpenSpaceToolkit/Physics/Environment.test.cpp:884]: Should be in the same function above in an additional scope

File: bindings/python/src/OpenSpaceToolkitPhysicsPy/Environment/Objects/CelestialBodies/Earth.cpp [hunk 8]
```diff
@@ -109,6 +113,18 @@ inline void                     OpenSpaceToolkitPhysicsPy_Environment_Objects_Ce
```
Comment (apaletta3) [inline bindings/python/src/OpenSpaceToolkitPhysicsPy/Environment/Objects/CelestialBodies/Earth.cpp:113]: These models are not properly attached to the class and can't be used in the bindings

File: bindings/python/src/OpenSpaceToolkitPhysicsPy/Environment/Gravitational/Earth.cpp [hunk 9]
```diff
@@ -18,6 +18,7 @@ inline void                     OpenSpaceToolkitPhysicsPy_Environment_Gravitatio
     using namespace pybind11 ;
 
     using ostk::core::fs::Directory ;
+    using ostk::core::types::Integer ;
```
Comment (Derollez) [inline bindings/python/src/OpenSpaceToolkitPhysicsPy/Environment/Gravitational/Earth.cpp:21]: add above `Directory`

File: src/OpenSpaceToolkit/Physics/Environment/Objects/CelestialBodies/Earth.cpp [hunk 10]
```diff
@@ -77,6 +77,16 @@ const Real                      Earth::Models::EGM96::C40
 const Real                      Earth::Models::EGM96::J2                        =       -Earth::Models::EGM96::C20 * std::sqrt(5.0) ;
 const Real                      Earth::Models::EGM96::J4                        =       -Earth::Models::EGM96::C40 * std::sqrt(9.0) ;
 
+// EGM84
+
+const Derived                   Earth::Models::EGM84::GravitationalParameter    =       { 398600441800000.0, GravitationalParameterSIUnit } ;
+const Length                    Earth::Models::EGM84::EquatorialRadius          =       Length::Meters(6378137.0) ;
+const Real                      Earth::Models::EGM84::Flattening                =       1.0 / 298.257223563 ;
+const Real                      Earth::Models::EGM84::C20                       =       -4.841668500000e-04 ; // TODO [TBR] Check to ensure that these C20 and C40 coeffs are correct
```
[the remark this answers is not in the document]
Comment (apaletta3) [reply src/OpenSpaceToolkit/Physics/Environment/Objects/CelestialBodies/Earth.cpp:85]: found the coefficients and added the link to the source of truth

File: README.md [hunk 11]
```diff
@@ -268,7 +268,7 @@ Or to run them manually:
 | Eigen         | 3.3.7      | MPL2                                                       | [eigen.tuxfamily.org](http://eigen.tuxfamily.org/index.php)                                                                                |
 | IAU SOFA      | 2018-01-30 | [SOFA Software License](http://www.iausofa.org/tandc.html) | [www.iausofa.org](http://www.iausofa.org)                                                                                                  |
 | SPICE Toolkit | N0066      | [NAIF](https://naif.jpl.nasa.gov/naif/rules.html)          | [naif.jpl.nasa.gov/naif/toolkit.html](https://naif.jpl.nasa.gov/naif/toolkit.html)                                                         |
-| GeographicLib | 1.49       | MIT                                                        | [geographiclib.sourceforge.io](https://geographiclib.sourceforge.io)                                                                       |
+| GeographicLib | 1.52       | MIT                                                        | [geographiclib.sourceforge.io](https://geographiclib.sourceforge.io)                                                                       |
```
Comment (Derollez) [inline README.md:271]: @apaletta3 Did you get a chance to dig into the following versions, what they had/fix potentially and the diff resulting in the errors we had seen when trying to install them? This is simply to ensure that:
- The following versions do not contain patches/fixes on top of `1.52` for instance. 
- We understand the changes made to that dependency which makes it incompatible with our current installation process today.

File: bindings/python/test/environment/__init__.py [hunk 12]
```diff
@@ -5,4 +5,15 @@
...
+
+import pytest
+
+import ostk.physics as physics
+
+################################################################################################################################################################
+
+
+
+
```
Comment (Derollez) [inline bindings/python/test/environment/__init__.py:18]:
```suggestion
```

File: bindings/python/test/environment/gravitational/earth/__init__.py [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+import pytest
+
+import ostk.physics as physics
+
+################################################################################################################################################################
+
+
+
+
+################################################################################################################################################################
```
Comment (Derollez) [inline bindings/python/test/environment/gravitational/earth/__init__.py:19]:
```suggestion
```

File: bindings/python/test/environment/gravitational/earth/test_earth_manager.py [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+import pytest
+import numpy as np
+
+from ostk.core.filesystem import Path
+from ostk.core.filesystem import Directory
+
+from ostk.io import URL
+
+from ostk.physics.time import Instant
+from ostk.physics.environment.objects.celestial_bodies import Earth
```
Comment (Derollez) [inline bindings/python/test/environment/gravitational/earth/test_earth_manager.py:19]:
```suggestion
```
Are these used in the code below? if not, TBR

File: bindings/python/test/environment/gravitational/earth/test_earth_manager.py [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+
+    assert manager.is_enabled() is not None
+
+    assert manager.has_data_file_for_type(EarthGravitationalModel.EarthType.WGS84) is not None
+
+    assert manager.get_local_repository() is not None
+
+    assert manager.get_remote_url() is not None
+
+    # manager.fetch_data_file_for_type(EarthGravitationalModel.EarthType.WGS84)
```
Comment (Derollez) [inline bindings/python/test/environment/gravitational/earth/test_earth_manager.py:49]: TBR?

File: bindings/python/test/environment/gravitational/earth/test_earth_manager.py [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+################################################################################################################################################################
+
+def test_earth_manager_methods ():
+
+    manager: EarthGravitationalModelManager = EarthGravitationalModelManager.get()
+
+    assert manager is not None
+    assert isinstance(manager, EarthGravitationalModelManager)
+
+    assert manager.is_enabled() is not None
```
Comment (Derollez) [inline bindings/python/test/environment/gravitational/earth/test_earth_manager.py:41]: assert expected value

File: bindings/python/test/environment/gravitational/earth/test_earth_manager.py [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+
+    assert manager.has_data_file_for_type(EarthGravitationalModel.EarthType.WGS84) is not None
+
+    assert manager.get_local_repository() is not None
```
Comment (Derollez) [inline bindings/python/test/environment/gravitational/earth/test_earth_manager.py:45]: same, assert and test the value expected

File: bindings/python/test/environment/gravitational/earth/test_earth_manager.py [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+
+    assert manager.get_remote_url() is not None
```
Comment (Derollez) [inline bindings/python/test/environment/gravitational/earth/test_earth_manager.py:47]: same

File: bindings/python/test/environment/gravitational/test_earth.py [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
+# @file           bindings/python/test/objects/gravitational/test_earth.py
+# @author         Antoine Paletta <antoine.paletta@loftorbital.com>
+# @license        Apache License 2.0
+
+################################################################################################################################################################
+
+import pytest
+import numpy as np
+
+from ostk.core.filesystem import Path
```
Comment (Derollez) [inline bindings/python/test/environment/gravitational/test_earth.py:13]: Same as the above, it seems `Path`, `URL`, `Earth` and `EarthGravitationalModelManager` are unused

File: bindings/python/test/environment/gravitational/earth/test_earth_manager.py [hunk 20]
```diff
@@ -0,0 +1,59 @@
+################################################################################################################################################################
+
+# @project        Open Space Toolkit ▸ Physics
+# @file           bindings/python/test/objects/gravitational/earth/test_earth_manager.py
```
Comment (Derollez) [inline bindings/python/test/environment/gravitational/earth/test_earth_manager.py:4]:
```suggestion
# @file           bindings/python/test/environment/gravitational/earth/test_earth_manager.py
```

File: bindings/python/test/environment/gravitational/test_earth.py [hunk 21]
```diff
@@ -0,0 +1,93 @@
+################################################################################################################################################################
+
+# @project        Open Space Toolkit ▸ Physics
+# @file           bindings/python/test/objects/gravitational/test_earth.py
```
Comment (Derollez) [inline bindings/python/test/environment/gravitational/test_earth.py:4]:
```suggestion
# @file           bindings/python/test/environment/gravitational/test_earth.py
```

File: bindings/python/test/environment/gravitational/test_earth.py [hunk 22]
```diff
@@ -0,0 +1,93 @@
...
+    assert earth_gravitational_model.get_type() is not None
+    assert earth_gravitational_model.get_type() == EarthGravitationalModel.EarthType.EGM2008
+
+################################################################################################################################################################
+
+def test_earth_get_field_value_at ():
+
+    earth_gravitational_model = construct_earth()
+
+    assert earth_gravitational_model.get_field_value_at(np.array([6400e3, 0.0, 0.0]), Instant.J2000()) is not None
```
Comment (Derollez) [inline bindings/python/test/environment/gravitational/test_earth.py:91]: assert value?

File: bindings/python/test/environment/gravitational/test_spherical.py [hunk 23]
```diff
@@ -0,0 +1,41 @@
+################################################################################################################################################################
+
+# @project        Open Space Toolkit ▸ Physics
+# @file           bindings/python/test/objects/gravitational/test_spherical.py
```
Comment (Derollez) [inline bindings/python/test/environment/gravitational/test_spherical.py:4]:
```suggestion
# @file           bindings/python/test/environment/gravitational/test_spherical.py
```

File: bindings/python/test/environment/gravitational/test_spherical.py [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+# @author         Antoine Paletta <antoine.paletta@loftorbital.com>
+# @license        Apache License 2.0
+
+################################################################################################################################################################
+
+import pytest
+import numpy as np
+
+from ostk.core.filesystem import Path
```
Comment (Derollez) [inline bindings/python/test/environment/gravitational/test_spherical.py:13]: `Path`, `Directory`, `Earth` and `URL` unused

File: bindings/python/test/environment/gravitational/test_spherical.py [hunk 25]
```diff
@@ -0,0 +1,41 @@
...
+    assert spherical_gravitational_model is not None
+    assert isinstance(spherical_gravitational_model, SphericalGravitationalModel)
+
+################################################################################################################################################################
+
+def test_spherical_get_field_value_at ():
+
+    spherical_gravitational_model: SphericalGravitationalModel = SphericalGravitationalModel(EGM2008.gravitational_parameter)
+
+    assert spherical_gravitational_model.get_field_value_at(np.array([6400e3, 0.0, 0.0]), Instant.J2000()) is not None
```
Comment (Derollez) [inline bindings/python/test/environment/gravitational/test_spherical.py:39]: use real value for assertion?

File: bindings/python/test/environment/objects/celestial/__init__ .py [hunk 26]
```diff
@@ -0,0 +1,19 @@
...
+import pytest
+
+import ostk.physics as physics
+
+################################################################################################################################################################
+
+
+
+
+################################################################################################################################################################
```
Comment (Derollez) [inline bindings/python/test/environment/objects/celestial/__init__ .py:19]:
```suggestion
```

File: bindings/python/test/environment/objects/celestial/test_earth.py [hunk 27]
```diff
@@ -0,0 +1,201 @@
+################################################################################################################################################################
+
+# @project        Open Space Toolkit ▸ Physics
+# @file           bindings/python/test/objects/celestial/test_earth.py
```
Comment (Derollez) [inline bindings/python/test/environment/objects/celestial/test_earth.py:4]: same, typo -> `environment` missing in path

File: bindings/python/test/environment/objects/celestial/test_earth.py [hunk 28]
```diff
@@ -0,0 +1,201 @@
...
+################################################################################################################################################################
+
+def test_earth_static_EGM2008_constructor ():
+
+    earth: Earth = Earth.EGM2008()
+
+    assert earth is not None
+    assert isinstance(earth, Earth)
+
+    earth: Earth = Earth.EGM2008(2190,2160)
```
Comment (Derollez) [inline bindings/python/test/environment/objects/celestial/test_earth.py:39]:
```suggestion
    earth: Earth = Earth.EGM2008(2190, 2160)
```
style

File: bindings/python/test/environment/objects/celestial/test_earth.py [hunk 29]
```diff
@@ -0,0 +1,201 @@
...
+################################################################################################################################################################
+
+def test_earth_static_EGM96_constructor ():
+
+    earth: Earth = Earth.EGM96()
+
+    assert earth is not None
+    assert isinstance(earth, Earth)
+
+    earth: Earth = Earth.EGM96(360,360)
```
Comment (Derollez) [inline bindings/python/test/environment/objects/celestial/test_earth.py:53]:
```suggestion
    earth: Earth = Earth.EGM96(360, 360)
```

File: bindings/python/test/environment/objects/celestial/test_earth.py [hunk 30]
```diff
@@ -0,0 +1,201 @@
...
+################################################################################################################################################################
+
+def test_earth_static_EGM84_constructor ():
+
+    earth: Earth = Earth.EGM84()
+
+    assert earth is not None
+    assert isinstance(earth, Earth)
+
+    earth: Earth = Earth.EGM84(180,180)
```
Comment (Derollez) [inline bindings/python/test/environment/objects/celestial/test_earth.py:67]:
```suggestion
    earth: Earth = Earth.EGM84(180, 180)
```

File: bindings/python/test/environment/objects/celestial/test_earth.py [hunk 31]
```diff
@@ -0,0 +1,201 @@
...
+################################################################################################################################################################
+
+def test_earth_static_WGS84_EGM96_constructor ():
+
+    earth: Earth = Earth.WGS84_EGM96()
+
+    assert earth is not None
+    assert isinstance(earth, Earth)
+
+    earth: Earth = Earth.WGS84_EGM96(180,180)
```
Comment (Derollez) [inline bindings/python/test/environment/objects/celestial/test_earth.py:81]:
```suggestion
    earth: Earth = Earth.WGS84_EGM96(180, 180)
```
and `earth` naming should not be re-used, give it a different name

File: bindings/python/test/environment/objects/celestial/test_earth.py [hunk 32]
```diff
@@ -0,0 +1,201 @@
...
+################################################################################################################################################################
+
+def test_earth_static_WGS84_constructor ():
+
+    earth: Earth = Earth.WGS84()
+
+    assert earth is not None
+    assert isinstance(earth, Earth)
+
+    earth: Earth = Earth.WGS84(20,0)
```
Comment (Derollez) [inline bindings/python/test/environment/objects/celestial/test_earth.py:95]:
```suggestion
    earth: Earth = Earth.WGS84(20, 0)
```
same on re-using `earth`

File: bindings/python/test/environment/objects/celestial/test_moon.py [hunk 33]
```diff
@@ -0,0 +1,23 @@
+################################################################################################################################################################
+
+# @project        Open Space Toolkit ▸ Physics
+# @file           bindings/python/test/objects/celestial/test_moon.py
```
Comment (Derollez) [inline bindings/python/test/environment/objects/celestial/test_moon.py:4]: missing `environment`

File: bindings/python/test/environment/objects/celestial/test_sun.py [hunk 34]
```diff
@@ -0,0 +1,23 @@
+################################################################################################################################################################
+
+# @project        Open Space Toolkit ▸ Physics
+# @file           bindings/python/test/objects/celestial/test_sun.py
```
Comment (Derollez) [inline bindings/python/test/environment/objects/celestial/test_sun.py:4]: missing `environment`

File: test/OpenSpaceToolkit/Physics/Environment.test.cpp [hunk 35]
```diff
@@ -30,6 +33,7 @@
 #include <OpenSpaceToolkit/Mathematics/Geometry/3D/Objects/Point.hpp>
 #include <OpenSpaceToolkit/Mathematics/Geometry/2D/Objects/Polygon.hpp>
 #include <OpenSpaceToolkit/Mathematics/Geometry/2D/Objects/Point.hpp>
+#include <OpenSpaceToolkit/Mathematics/Objects/Vector.hpp>
```
Comment (Derollez) [inline test/OpenSpaceToolkit/Physics/Environment.test.cpp:36]: Purpose of these extra imports?

File: Makefile [hunk 36]
```diff
@@ -143,6 +143,8 @@ _build-development-image: _pull-development-image
 	--build-arg="VERSION=$(docker_image_version)" \
 	"$(project_directory)"
 
+
+
```
Comment (lucas-bremond) [inline Makefile:147]: TBR?

File: bindings/python/test/environment/__init__.py [hunk 37]
```diff
@@ -6,3 +6,9 @@
 # @license        Apache License 2.0
 
 ################################################################################################################################################################
+
+import pytest
+
+import ostk.physics as physics
+
+################################################################################################################################################################
```
Comment (lucas-bremond) [inline bindings/python/test/environment/__init__.py:14]: Is this necessary?

File: bindings/python/test/environment/gravitational/earth/__init__.py (hunk 13 above)
Comment (lucas-bremond) [reply bindings/python/test/environment/gravitational/earth/__init__.py:19]: Why was this resolved? I don't think we should keep this.

File: bindings/python/test/environment/objects/celestial/__init__ .py (hunk 26 above)
Comment (lucas-bremond) [reply bindings/python/test/environment/objects/celestial/__init__ .py:19]: Same

File: docker/development/debian/Dockerfile (hunk 1 above)
Comment (lucas-bremond) [inline docker/development/debian/Dockerfile:88]:
```suggestion
# GeographicLib [1.52]
```

for consistency

File: docker/development/debian/Dockerfile [hunk 38]
```diff
@@ -127,6 +127,14 @@ RUN mkdir -p /tmp/open-space-toolkit-mathematics \
  && apt-get install -y ./*.deb \
  && rm -rf /tmp/open-space-toolkit-mathematics
 
+
```
Comment (lucas-bremond) [inline docker/development/debian/Dockerfile:130]: Extra newline

File: include/OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.hpp [hunk 39]
```diff
@@ -60,20 +63,25 @@ class Earth : public Model
...
         /// @param              [in] (optional) aDataDirectory A gravitational model data directory
+        /// @param              [in] (optional) aGravityModelDegree A gravitational model degree
+        /// @param              [in] (optional) aGravityModelOrder A gravitational model order
 
                                 Earth                                       (   const   Earth::Type&                aType,
-                                                                                const   Directory&                  aDataDirectory                              =   Directory::Undefined() ) ;
+                                                                                const   Directory&                  aDataDirectory                              =   Directory::Undefined(),
+                                                                                const   Integer&                    aGravityModelDegree                         = -1,
+                                                                                const   Integer&                    aGravityModelOrder                          = -1    ) ;
```
Comment (lucas-bremond) [inline include/OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.hpp:84]: Instead of using the special value `-1` to convey that a value is undefined, use `Integer::Undefined()`.

File: include/OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.hpp [hunk 40]
```diff
@@ -34,11 +34,14 @@ namespace gravitational
...
 
 using ostk::core::types::Unique ;
+using ostk::core::types::Integer ;
 using ostk::core::types::Real ;
 using ostk::core::fs::Directory ;
 
 using ostk::io::URL ;
 
+using ostk::physics::units::Length ;
+using ostk::physics::units::Time ;
```
Comment (lucas-bremond) [inline include/OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.hpp:44]: Why is this necessary here?

**Review by lucas-bremond — CHANGES_REQUESTED** [review]

File: bindings/python/test/environment/__init__.py (hunk 37 above)
Comment (Derollez) [reply bindings/python/test/environment/__init__.py:14]: I don't think it is. @apaletta3 Did you apply the previous suggestions to remove this? 🤔 (I might have missed a file or 2 when adding those suggestions)

File: bindings/python/test/environment/gravitational/test_earth.py [hunk 41]
```diff
@@ -0,0 +1,90 @@
...
+
+################################################################################################################################################################
+
+def test_earth_get_field_value_at ():
+
+    earth_gravitational_model = construct_earth()
+
+    grav_acceleration = earth_gravitational_model.get_field_value_at(np.array([6400e3, 0.0, 0.0]), Instant.J2000())
+    grav_acceleration_ref = np.array([-9.74722832e+00, -1.29220148e-05, -1.34173896e-05])
+    assert all( [ round(grav_acceleration[i], -7) == round(grav_acceleration_ref[i], -7) for i in range(0, 2) ] )
```
Comment (Derollez) [inline bindings/python/test/environment/gravitational/test_earth.py:88]:
```suggestion
    assert all([round(grav_acceleration[i], -7) == round(grav_acceleration_ref[i], -7) for i in range(0, 2) ])
```

File: bindings/python/test/environment/gravitational/test_spherical.py [hunk 42]
```diff
@@ -0,0 +1,37 @@
...
+
+################################################################################################################################################################
+
+def test_spherical_get_field_value_at ():
+
+    spherical_gravitational_model: SphericalGravitationalModel = SphericalGravitationalModel(EGM2008.gravitational_parameter)
+
+    grav_acceleration = spherical_gravitational_model.get_field_value_at(np.array([6400e3, 0.0, 0.0]), Instant.J2000())
+    grav_acceleration_ref = np.array([-9.73145609, 0, 0])
+    assert all( [ round(grav_acceleration[i], -7) == round(grav_acceleration_ref[i], -7) for i in range(0, 2) ] )
```
Comment (Derollez) [inline bindings/python/test/environment/gravitational/test_spherical.py:35]: same

File: bindings/python/test/environment/objects/celestial/__init__ .py [hunk 43]
```diff
@@ -0,0 +1,19 @@
...
+
+# @project        Open Space Toolkit ▸ Physics
+# @file           bindings/python/test/environment/objects/celestial/__init__.py
+# @author         Antoine Paletta <antoine.paletta@loftorbital.com>
+# @license        Apache License 2.0
+
+################################################################################################################################################################
+
+import pytest
```
Comment (Derollez) [inline bindings/python/test/environment/objects/celestial/__init__ .py:10]: TBR, I don't think imports are necessary here (see previous suggestions)

File: bindings/python/test/environment/gravitational/earth/__init__.py (hunk 13 above)
Comment (Derollez) [reply bindings/python/test/environment/gravitational/earth/__init__.py:19]: @apaletta3

File: Makefile (hunk 36 above)
Comment (apaletta3) [reply Makefile:147]: addressed

File: bindings/python/test/environment/__init__.py (hunk 37 above)
Comment (apaletta3) [reply bindings/python/test/environment/__init__.py:14]: fixed

File: docker/development/debian/Dockerfile (hunk 1 above)
Comment (apaletta3) [reply docker/development/debian/Dockerfile:88]: fixed

File: docker/development/debian/Dockerfile (hunk 38 above)
Comment (apaletta3) [reply docker/development/debian/Dockerfile:130]: fixed

File: include/OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.hpp (hunk 39 above)
Comment (apaletta3) [reply include/OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.hpp:84]: @lucas-bremond @Derollez There is a specific reason why I used -1 instead of an undefined integer, because Geographic lib interprets negative numbers a specific way and does not truncate the gravity model. If you inputted an undefined integer it would break how geographic lib interprets the numbers.

Comment (apaletta3) [reply include/OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.hpp:84]: https://github.com/geographiclib/geographiclib/blob/main/include/GeographicLib/GravityModel.hpp#L193

File: include/OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.hpp (hunk 40 above)
Comment (apaletta3) [reply include/OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.hpp:44]: fixed

File: bindings/python/test/environment/gravitational/test_earth.py (hunk 41 above)
Comment (apaletta3) [reply bindings/python/test/environment/gravitational/test_earth.py:88]: fixed

File: bindings/python/test/environment/gravitational/test_spherical.py (hunk 42 above)
Comment (apaletta3) [reply bindings/python/test/environment/gravitational/test_spherical.py:35]: fixed

File: bindings/python/test/environment/objects/celestial/__init__ .py (hunk 43 above)
Comment (apaletta3) [reply bindings/python/test/environment/objects/celestial/__init__ .py:10]: fixed

File: bindings/python/test/environment/gravitational/test_earth.py [hunk 44]
```diff
@@ -0,0 +1,90 @@
...
+    assert earth_gravitational_model.get_type() == EarthGravitationalModel.EarthType.EGM2008
+
+################################################################################################################################################################
+
+def test_earth_get_field_value_at ():
+
+    earth_gravitational_model = construct_earth()
+
+    grav_acceleration = earth_gravitational_model.get_field_value_at(np.array([6400e3, 0.0, 0.0]), Instant.J2000())
+    grav_acceleration_ref = np.array([-9.74722832e+00, -1.29220148e-05, -1.34173896e-05])
```
Comment (vishwaLoft) [inline bindings/python/test/environment/gravitational/test_earth.py:87]: Can you put a comment here as to where you get this value from?
Incase we ever need to update the unit test in the future that will be useful to know.

File: bindings/python/test/environment/gravitational/test_earth.py [hunk 45]
```diff
@@ -0,0 +1,90 @@
...
+    assert all([round(grav_acceleration[i], -7) == round(grav_acceleration_ref[i], -7) for i in range(0, 2)])
```
Comment (vishwaLoft) [inline bindings/python/test/environment/gravitational/test_earth.py:88]:
```suggestion
    assert all([round(grav_acceleration[i], -7) == round(grav_acceleration_ref[i], -7) for i in range(0, len(grav_acceleration_ref))])
```

File: bindings/python/test/environment/gravitational/test_spherical.py [hunk 46]
```diff
@@ -0,0 +1,37 @@
...
+    assert isinstance(spherical_gravitational_model, SphericalGravitationalModel)
+
+################################################################################################################################################################
+
+def test_spherical_get_field_value_at ():
+
+    spherical_gravitational_model: SphericalGravitationalModel = SphericalGravitationalModel(EGM2008.gravitational_parameter)
+
+    grav_acceleration = spherical_gravitational_model.get_field_value_at(np.array([6400e3, 0.0, 0.0]), Instant.J2000())
+    grav_acceleration_ref = np.array([-9.73145609, 0, 0])
```
Comment (vishwaLoft) [inline bindings/python/test/environment/gravitational/test_spherical.py:34]: Same as above.

**Review by vishwaLoft — COMMENTED** [review]
This looks great man! Excellent work with the unit tests.

File: bindings/python/test/environment/gravitational/test_earth.py (hunk 44 above)
Comment (apaletta3) [reply bindings/python/test/environment/gravitational/test_earth.py:87]: Since I know that the gravity model is currently working, I just printed out the value and put that in the unit test. @vishwaLoft

File: bindings/python/test/environment/objects/celestial/test_earth.py [hunk 47]
```diff
@@ -0,0 +1,201 @@
...
+    earth: Earth = Earth.spherical()
+
+    assert earth is not None
+    assert isinstance(earth, Earth)
+
+################################################################################################################################################################
+
+def test_earth_static_properties_EGM2008 ():
+
+    assert EGM2008.gravitational_parameter is not None
```
Comment (apaletta3) [inline bindings/python/test/environment/objects/celestial/test_earth.py:113]: These are left as `is not None` because the values are hard coded in the Earth.cpp file so I don't think it makes sense to have them hardcoded here in a unit test.

File: src/OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.cpp [hunk 48]
```diff
@@ -86,6 +99,18 @@ Vector3d                        Earth::Impl::getFieldValueAt                (
 
     (void) anInstant ; // Temporal invariance
 
+    if (type_ == Earth::Type::Spherical)
```
Comment (apaletta3) [inline src/OpenSpaceToolkit/Physics/Environment/Gravitational/Earth.cpp:102]: This was done here instead of calling Earth::Spherical because something is partially broken in Earth::Spherical when you construct it from the Earth.cpp file, so its simpler to calculate the Two Body gravity as this.

**Review by apaletta3 — COMMENTED** [review]
One more change to be made cause the rebase changed the shared folder internal file structure

File: bindings/python/test/environment/gravitational/test_earth.py (hunk 44 above)
Comment (vishwaLoft) [reply bindings/python/test/environment/gravitational/test_earth.py:87]: Hmmmmmm fair enough. So it's basically just self-validating the function. I thought for some reason you were checking against values from GMAT or sth.

**Review by lucas-bremond — APPROVED** [review]

**Review by Derollez — APPROVED** [review]
